html/semantics/forms/form-submission-0/form-submission-algorithm.html

```diff
@@ -50,6 +50,33 @@
   });
 }, "If form's firing submission events is true, then return; 'submit' event");
 
+promise_test(async () => {
+  let form = populateForm('<button type=submit></button><input name=n1 value=submit type=submit>');
+  let iframe = form.previousSibling;
+  let submitter = form.querySelector('input[type=submit]');
+  let event;
+  form.addEventListener('submit', e => { event = e; });
+  submitter.click();
+  await loadPromise(iframe);
+  assert_true(event.bubbles);
+  assert_true(event.cancelable);
+  assert_equals(event.submitter, submitter);
+  assert_true(event instanceof SubmitEvent);
+}, 'firing an event named submit; clicking a submit button');
+
+promise_test(async () => {
+  let form = populateForm('');
+  let iframe = form.previousSibling;
+  let event;
+  form.addEventListener('submit', e => { event = e; });
+  form.requestSubmit();
+  await loadPromise(iframe);
+  assert_true(event.bubbles);
+  assert_true(event.cancelable);
+  assert_equals(event.submitter, form);
+  assert_true(event instanceof SubmitEvent);
+}, 'firing an event named submit; form.requestSubmit()');
```

Comment:
We should add a test with requestSubmit(submitter)

Reply:
Done.